src/Resolvers/DigiDSegmentResolver.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OWC\Zaaksysteem\Resolvers;
+
+use Aura\Session\Segment;
+
+use function Yard\DigiD\Foundation\Helpers\decrypt;
+use function Yard\DigiD\Foundation\Helpers\resolve;
+
+class DigiDSegmentResolver
+{
+    protected Segment $segment;
+
+    final private function __construct()
+    {
+        $this->segment = resolve('session')->getSegment('digid');
+    }
+
+    public static function make(): self
+    {
+        return new static();
+    }
+
+    public function get(string $key)
```

Comment:
Kan deze ge-typehint worden met iets of kan er van alles in zo'n segment zitten? 

Belangrijker nog: ga je er ooit iets anders uit halen dan het BSN nummer? Misschien zou de `get()` weg kunnen of een synoniem zijn voor `bsn()`.

Reply:
Op dit moment zou dat wel kunnen, heb het er zo in gezet omdat we misschien ooit wel meer willen ophalen dan alleen het BSN nummer. Mij maakt het niet zoveel uit eigenlijk.

Ik pas de bsn method aan naar:
```
  public function bsn(): string
  {
      $bsn = $this->get('bsn');

      return ! empty($bsn) && is_string($bsn) ? decrypt($bsn) : '';
  }
```